### ISSUE TITLE
Fever reports large deficits for buildings, unaffected by insulation

### ISSUE THREAD

**Issue by michieldenhaan** [quintel/etmodel#2954]
Scenario with 100% heat pumps in Buildings. Fever reports large deficits in February and December:

![beta-pro energytransitionmodel com - energy transition model - your free independent comprehensive fact-based scenario build 2019-02-08 14-27-03](https://user-images.githubusercontent.com/32056448/52481290-6167b080-2bae-11e9-8137-44320050f17e.png)

![beta-pro energytransitionmodel com - energy transition model - your free independent comprehensive fact-based scenario build 2019-02-08 14-27-13](https://user-images.githubusercontent.com/32056448/52481342-89efaa80-2bae-11e9-807c-3d6a4c7c0950.png)

When increasing insulation levels absolute heat demand decreases but deficits stay the same (relatively):

![beta-pro energytransitionmodel com - energy transition model - your free independent comprehensive fact-based scenario build 2019-02-08 14-27-27](https://user-images.githubusercontent.com/32056448/52481451-db983500-2bae-11e9-855e-01910b91b911.png)

![beta-pro energytransitionmodel com - energy transition model - your free independent comprehensive fact-based scenario build 2019-02-08 14-27-37](https://user-images.githubusercontent.com/32056448/52481460-e357d980-2bae-11e9-9bcb-e348544b5b75.png)

I think this is because:

- Total installed heat pump capacity is based on total year demand
- Insulation decreases heat demand and hence installed capacity but does not change the _shape_ of the demand profile
- So the demand peaks relative to installed capacity stay the same: the reduction in heat demand due to insulation is relatively the same as the reduction in installed capacity due to insulation

To sum up, there's two issues here:

- The heat demand profile is 'too peaky' to begin with (or at least 'too peaky' relative to installed capacity), leading to very large deficits before insulation
- The heat demand profile's shape does not change with insulation, causing these deficits to persist after insulation

Notifying @AlexanderWirtz, @ChaelKruip, @marliekeverweij

**Comment by ChaelKruip** [comment]
> The heat demand profile is 'too peaky' to begin with (or at least 'too peaky' relative to installed capacity), leading to very large deficits before insulation

Yes. This is difficult to fix quickly as we need to research better profiles for this.

> The heat demand profile's shape does not change with insulation, causing these deficits to persist after insulation

This should not be a real issue as long as the relative hight of the peaks with respect to the *installed* capacity of the technologies is reduced. This is currently not the case because the *number of units* of the technologies scale with the heat demand in buildings as well.

This is different from how the heat producers work in households: every household gets exactly one unit and the full_load_hours are flexible instead of the number_of_units.

One way to go about this is to make heat producers in buildings behave similarly: relating their number_of_units to the number of square meters of buildings (instead of heat demand). 

@michieldenhaan @marliekeverweij do we know the number of square meters of buildings? Related question: what does the number_of_buildings attribute really mean? Is the is the *real* number of buildings or a number of 'standard QI buildings" which have fixed size. I remember we used to have a "standard QI building" defined for this purpose right @AlexanderWirtz?

**Comment by ChaelKruip** [comment]
Adding this to deploy milestone.

**Comment by ChaelKruip** [comment]
@antw can you explain how the number of units and the full load hours of heating technologies in buildings are currently set?

@marliekeverweij and I expect the number_of_units to be set by the demand_driven method: [https://github.com/quintel/etengine/blob/master/app/models/qernel/demand_driven_converter_api.rb#L49-L71](https://github.com/quintel/etengine/blob/master/app/models/qernel/demand_driven_converter_api.rb#L49-L71) but this explicitly refers to `number_of_residences`...

**Comment by antw** [comment]
> @antw can you explain how the number of units and the full load hours of heating technologies in buildings are currently set?

The building heating converters are not currently `demand_driven`, so ETEngine calculates their number of units the same way it does any other converter:

```ruby
demand / (input_capacity * full_load_seconds)
```

This is likely why there is a deficit: ETEngine is installing fewer heat pumps that you would expect from the slider setting.

**Comment by ChaelKruip** [comment]
> The building heating converters are not currently `demand_driven`, so ETEngine calculates their number of units the same way it does any other converter

I am confused by this. Wouldn't you expect the `full_load_seconds` to be static in that case? 

If in a blank NL 2050 scenario, I set the `buildings_space_heater_heatpump_air_water_network_gas` to 100%, I see both its number of units (expected) and its `full_load_seconds` (unexpected!) increase:


Label | Present | Future
-- | -- | --
Full load hours | 4,380 | 4,167.7
number_of_units | 0 | 4,168.1
-- | -- | --

**Comment by antw** [comment]
> Wouldn't you expect the `full_load_seconds` to be static in that case?

Fever updates the `full_load_seconds` based on the time-resolved load.

It's relatively easy to make the building heaters demand-driven, but it may need some research and tweaking. Making the converters demand-driven completely eliminates the deficit when using heat pumps, however the default scenario now has a larger deficit due to there being fewer network gas heaters (346k vs 422k previously).

| Beta | Local, demand-driven |
| :---: | :---: |
| <img width="617" alt="screen shot 2019-02-12 at 13 55 52" src="https://user-images.githubusercontent.com/4383/52640240-f845ad00-2ecd-11e9-8540-06ae5d063e10.png"> | <img width="609" alt="screen shot 2019-02-12 at 13 56 00" src="https://user-images.githubusercontent.com/4383/52640260-ff6cbb00-2ecd-11e9-9980-c9b36a980081.png"> |
